apps/src/gamelab/GameLab.js

```diff
@@ -177,6 +216,34 @@ GameLab.prototype.init = function (config) {
   });
 };
 
+function quote(str) {
+  return '"' + str + '"';
+}
+
+/**
+ * Returns a list of options (optionally filtered by type) for code-mode
+ * asset dropdowns.
+ */
+GameLab.prototype.getAssetDropdown = function (typeFilter) {
+  var options = assetListStore.list(typeFilter).map(function (asset) {
+    return {
+      text: quote(asset.filename),
+      display: quote(asset.filename)
+    };
+  });
+  var handleChooseClick = function (callback) {
+    showAssetManager(function (filename) {
+      callback(quote(filename));
+    }, typeFilter);
+  };
+  options.push({
+    text: 'Choose...',
+    display: '<span class="chooseAssetDropdownOption">Choose...</a>',
+    click: handleChooseClick
+  });
+  return options;
+};
+
```

Comment:
Instead of extracting these into `StudioApp` (already a god-class in its own right) maybe we can create a common AssetManager thing that owns asset-related common code.